e2e-tests/adapters/cypress/e2e/headers.cy.ts

```diff
@@ -0,0 +1,54 @@
+import { title } from "../../constants"
+
+describe("Headers", () => {
+  function checkHeaders(routeAlias, expectedHeaders) {
+    cy.wait(routeAlias).then(interception => {
+      Object.keys(expectedHeaders).forEach(headerKey => {
+        expect(interception.response.headers[headerKey]).to.eq(
+          expectedHeaders[headerKey]
+        )
+      })
+    })
+  }
+
+  const defaultHeaders = {
+    "x-xss-protection": "1; mode=block",
+    "x-content-type-options": "nosniff",
+    "referrer-policy": "same-origin",
+    "x-frame-options": "DENY",
+  }
+
+  it("should contain correct headers for index page", () => {
+    cy.intercept("/").as("index")
+    cy.visit("/")
+
+    checkHeaders("@index", {
+      ...defaultHeaders,
+      "x-custom-header": "my custom header value",
+    })
+  })
+
+  //   it("should contain correct headers for static assets", () => {
+  //     @todo
```

Comment:
I was struggling with getting this test to work with the immutable header. Open to suggestions here!

Reply:
Not sure if you hit same issue - but I did hit https://docs.cypress.io/api/commands/intercept#cyintercept-and-request-caching and used second workaround (first is absolutely not what we want, and third didn't seem to work for me).

https://github.com/gatsbyjs/gatsby/blob/9a267006a7ed8fe7bcd7773c5aab5f015a4f1449/e2e-tests/adapters/cypress/e2e/basics.cy.ts#L6-L15

If the troubles are because of the same issue - maybe we can use similar approach but additionally capture original `cache-control` as another header to be able to actually assert that one? maybe some helper to revert the response headers values to actual ones before asserting on them?